packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java

```diff
@@ -299,9 +303,47 @@ public boolean onActivityResult(int requestCode, int resultCode, Intent data) {
   }
 
   private void handleChoosePictureResult(int resultCode, Intent data) {
-    if (resultCode == Activity.RESULT_OK && data != null) {
+    if (resultCode == Activity.RESULT_OK && data != null && data.getData() != null) {
+      //Extract the file path from the uri.
       String path = fileUtils.getPathFromUri(activity, data.getData());
-      handleResult(path);
+      //Path may be null if, for example, source is remote.
+      if (path == null) {
+        InputStream inputStream = null;
+        OutputStream outputStream = null;
+        try {
+          inputStream = activity.getContentResolver().openInputStream(data.getData());
+          File file = File.createTempFile("cachedImage", "jpg", activity.getCacheDir());
```

Comment:
Just thinking if `cachedImage.jpg` is a too generic name for an image to be stored in a cache directory. There could be a chance we're overwriting some other image by accident. Would it make sense to use something less common, like `flutter_image_picker_cached_remote_image.jpg` for example?

UUIDs would be another option, but that could blow up the application's cache directory over time since images can get quite big. This might potentially give end users a bad impression of the app that uses this plugin.

Reply:
@roughike normay using createTempFile automatic append a random numeric string to the end of the name so that files will be "cachedImage-123456789.jpg" or something similar. I did not find of a nice way of automatically deleting the file created in the cacheDir